datafusion-examples/examples/advanced_udf.rs

```diff
@@ -215,9 +265,29 @@ async fn main() -> Result<()> {
     // print the results
     df.show().await?;
 
-    // You can also invoke both pow(2, 10)  and its alias my_pow(a, b) using SQL
-    let sql_df = ctx.sql("SELECT pow(2, 10), my_pow(a, b) FROM t").await?;
-    sql_df.show().await?;
+    // You can also invoke both pow(2, 10) and its alias my_pow(a, b) using SQL
+    ctx.sql("SELECT pow(2, 10), my_pow(a, b) FROM t")
+        .await?
+        .show()
+        .await?;
+
+    // You can also invoke pow_in_place by passing a constant base and a
+    // column `a` as the exponent . If there is only a single
+    // reference to `a` the code works well
+    ctx.sql("SELECT pow(2, a) FROM t").await?.show().await?;
+
+    // However, if there are multiple references to `a` in the evaluation
+    // the array storage can not be reused
+    let err = ctx
+        .sql("SELECT pow(2, a), pow(3, a) FROM t")
+        .await?
+        .show()
+        .await
+        .unwrap_err();
+    assert_eq!(
+        err.to_string(),
+        "Execution error: Could not reuse array for pow_in_place"
+    );
```

Comment:
This are good examples

Reply:
Went to test if this will work currently, found a bug in existing code :)

```sql
> SELECT pow(2, a), pow(3, a) FROM t;
Arrow error: Arithmetic overflow: Overflow happened on: 3 ^ 41
```

Reply:
... and the example sql does indeed work currently (as expected) so this would be a breaking change if it were to fail now.

Reply:
cool -- I will respond to these comments later today. Thank you